### PR TITLE
PHP 8: Code Review `Wiki` Part III

### PR DESCRIPTION
**Pull request by mjansenDatabay**
Hi @alex40724,

I completed the 3rd part of the review of the `Modules/Wiki` component.

Results:
- [x] The code style is `PSR-2 + X` compliant: 1 File has been changed by the `PHP-CS-FIXER`
- [x] No usage of `$_GET` / `$_POST` / `$_REQUEST` / `$_SESSION`
- [x] There is a `Unit Test Suite` with at least one unit test
- [x] The unit tests pass
- [x] External libraries are compatible with PHP 8 (if relevant)
- [ ] There are no serious `PhpStorm Code Inspection` issues left
- [ ] The types are fully documented (PhpDoc) or explict PHP types are used (type hints, return types, typed properties)

I also fixed some trivial issues reported by my inspection profile.

--

Actions needed:

- [x] I am not sure whether the libraries `Sanitize.php` and `Title.php` are compatible with PHP 8. Please check the list item if you already checked this.
- [ ] Please check my inline comments. Some of them are just information, some require an action. You can use `TODO PHP8-REVIEW` when searching.

Best regards,
@mjansenDatabay

### CODE REVIEW

File: Modules/Wiki/classes/class.ilObjWiki.php [hunk 1]
```diff
@@ -769,13 +769,13 @@ public function createWikiPage(
         return $page;
     }
 
-    public static function getAdvMDSubItemTitle($a_obj_id, $a_sub_type, $a_sub_id)
+    public static function getAdvMDSubItemTitle($a_obj_id, $a_sub_type, $a_sub_id) : string // TODO PHP8-REVIEW Type hints are missing here
```
Comment (mjansenDatabay) [inline Modules/Wiki/classes/class.ilObjWiki.php:772]: Type hints are missing here

File: Modules/Wiki/classes/class.ilWikiStat.php [hunk 2]
```diff
@@ -758,15 +758,15 @@ protected static function getWikiAggrSub(
...
+                        $res[$day] = array_sum($values) / count($values);
                         break;
                     
                     case "SUM(%s)":
                         $res[$day] = array_sum($values);
                         break;
                     
                     default:
-                        var_dump("unsupport aggr " . $a_aggr_value);
+                        var_dump("unsupport aggr " . $a_aggr_value);// TODO PHP8-REVIEW We should not use debugging code here
```
Comment (mjansenDatabay) [inline Modules/Wiki/classes/class.ilWikiStat.php:769]: We should not use debugging code here

File: Modules/Wiki/classes/class.ilWikiUtil.php [hunk 3]
```diff
@@ -447,7 +446,7 @@ public static function splitTrail(
 
     public static function sendNotification(
         string $a_action,
-        string $a_type,
+        string $a_type, // TODO PHP8-REVIEW Type is a `string` but is passed to methods expecting an `int`
```
Comment (mjansenDatabay) [inline Modules/Wiki/classes/class.ilWikiUtil.php:449]: Type is a `string` but is passed to methods expecting an `int`

File: Modules/Wiki/classes/class.ilObjWikiGUI.php [hunk 4]
```diff
@@ -1631,7 +1629,7 @@ public function exportHTML() : void
      * Get title for wiki page (used in ilNotesGUI)
      */
     public static function lookupSubObjectTitle(
-        string $a_wiki_id,
+        string $a_wiki_id,// TODO PHP8-REVIEW This is a `string`, but it is passed to a constructor expecting and `int`. Furthermore it is compared to an `int` below.
```
Comment (mjansenDatabay) [inline Modules/Wiki/classes/class.ilObjWikiGUI.php:1632]: This is a `string`, but it is passed to a constructor expecting and `int`. Furthermore it is compared to an `int` below.

File: Modules/Wiki/classes/class.ilObjWiki.php (hunk 1 above)
Comment (alex40724) [reply Modules/Wiki/classes/class.ilObjWiki.php:772]: They are missing in the metadata service.

**Comment by alex40724** [comment]
Fixed everything, except the metadata call, which needs to be fixed in Services/Medata...

**Pull request merged**